### ISSUE TITLE
Codestream broken in Clion 2021.1.1

### ISSUE THREAD

**Issue by RoscoeTheDog** [TeamCodeStream/codestream#535]
Hey everyone. This plugin looks really great! I was really was looking forward to trying it out but it seems it is broken with the latest version of Clion 2021.1.1. 

When I first signed in, the window pane came up and I linked my Github account. It seemed to give me a nice context menu and all was well. I tried to create my first comment on a chunk of code and it gave me a new window prompt saying an error had occurred. It said something like the project root directory and then _"No action"_. Unfortunately I cannot reproduce this error since now a bigger problem has shown it's ugly face. When I restarted CLion, the window pane just appears blank as seen below:

![image](https://user-images.githubusercontent.com/54086262/116798745-be115780-aaa7-11eb-98c9-1d66d791acc6.png)

I went into the CLion settings menu, found the plugin settings and turned off auto-sign in to see if that would prompt me to the first window I was in when the plug first initialized but this did not work. I even tried uninstalling and reinstalling the plugin with no success or change in behavior. It just seems that it's stuck on an empty menu now.

**Comment by RoscoeTheDog** [comment]
I didn't notice in the tool tray that there was a debug logger. Here is what is displayed:

```
kotlin.UninitializedPropertyAccessException: lateinit property webView has not been initialized
	at com.codestream.webview.WebViewService.getWebView(WebViewService.kt:33)
	at com.codestream.webview.CodeStreamToolWindowFactory$createToolWindowContent$1$1.run(CodeStreamToolWindowFactory.kt:32)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:218)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:200)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:319)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:839)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:808)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by planteater** [comment]
@RoscoeTheDog - Could I trouble you for a log file? Reproduce the issue and then go to Help > Collect Logs and Diagnostic Data. This will open the finder where you should see a newly created zip file.

if you’d prefer not to post the file here you can email it to me at support@codestream.com. Thanks!

**Comment by planteater** [comment]
@RoscoeTheDog - Below is a pre-release version of the CodeStream extension that I hope will address your issue.

https://www.dropbox.com/s/evirtjfnlej1w1o/codestream-jb-11.0.5%2B4121.zip?dl=0

Once you've downloaded the file, you can install it by going to Preferences/Settings > Plugins and then select "Install Plugin from Disk" from the gear menu at the top-right. You'll then be prompted to restart your IDE.

Let me know how it goes! Thanks.

**Comment by RoscoeTheDog** [comment]
Thanks for the swift response. The update seemed to unbreak the completely blank pane menu screen. But I cannot do anything without getting this git error "invalid expression":

![image](https://user-images.githubusercontent.com/54086262/116946669-9a791900-ac2f-11eb-9737-1b4324a15c4a.png)

[clion-logs-20210503-165110.zip](https://github.com/TeamCodeStream/codestream/files/6418393/clion-logs-20210503-165110.zip)

As requested, I have uploaded the logs for your review.

I am not familiar at all with the source code of this plugin, but could it possibly be a pathing issue such as special characters not being escaped? I have git/github installed already, I don't see how it could be an issue with dependencies but maybe there's something I don't know of needed for this plugin...

**Comment by planteater** [comment]
One more build for you to try @RoscoeTheDog. Thanks!

https://www.dropbox.com/s/ht1yw1ecqfxc8xg/codestream-jb-11.0.5%2B4126.zip?dl=0

**Comment by RoscoeTheDog** [comment]
Hehe I hope you guys like challenges..! Looks like it's still bugged unfortunately :(

This time I tried installing the updated zip. When that didn't work I uninstalled it completely and reinstalled just using the zip. I wanted to rule out that patching over the previous builds were not interfering with any potential fixes you might have been working on.

![image](https://user-images.githubusercontent.com/54086262/116952807-6f4af580-ac40-11eb-9da0-8f5579b47e87.png)

Here is the updated log.
[clion-logs-20210503-185521.zip](https://github.com/TeamCodeStream/codestream/files/6418702/clion-logs-20210503-185521.zip)

**Comment by planteater** [comment]
Here's another build. Fingers crossed. ;-)

https://www.dropbox.com/s/ua4vnnru9gzm379/codestream-jb-11.0.5%2B4132.zip?dl=0

**Comment by RoscoeTheDog** [comment]
Oh boy! I am wagging my tail right now. Whatever you wizards did, it is fixed! Whoever solved that bug deserves a beer 😄 🍺

![image](https://user-images.githubusercontent.com/54086262/117214640-07b4b780-adb2-11eb-88fb-4a591c42d318.png)

Thanks for working that out. Happy to break in this plugin a bit more now 👍

**Comment by planteater** [comment]
Excellent! Glad to hear it!

**Comment by marcelofarias** [comment]
> Oh boy! I am wagging my tail right now. Whatever you wizards did, it is fixed! Whoever solved that bug deserves a beer 😄 🍺
@RoscoeTheDog A Hazy IPA will do it

**Comment by RoscoeTheDog** [comment]
> > Oh boy! I am wagging my tail right now. Whatever you wizards did, it is fixed! Whoever solved that bug deserves a beer 😄 🍺
> > @RoscoeTheDog A Hazy IPA will do it

We got plenty of that over here in Oregon lol. Where do I send a tip?

**Comment by planteater** [comment]
We'll take you up on that next time we're out in Oregon. In the meantime, you should check out an episode of CodeStream On Tap! https://www.codestream.com/on-tap

**Comment by planteater** [comment]
V11.0.5 is now available in the marketplace.

**Comment by iLeW** [comment]
Hello all, I know this is a closed issue but I have the (almost) same problem here. So, before opening a new issue I would like to point out that after installing the CodeStream plugin (both on CLion 2021.2.1 and PyCharm 2021.2.1 on Arch Linux OS) my CodeStream tab is empty directly from a fresh install.

CodeStream version 11.0.14+186.

@planteater any cues?

**Comment by planteater** [comment]
@iLeW - Unfortunately, I think it's this issue:

https://github.com/TeamCodeStream/codestream/issues/642

The problem is with JetBrain's JCEF webview rendering, but look for the reply in that thread from marcelofarias for a workaround.

**Comment by iLeW** [comment]
Thank you @planteater, solved thanks to #642 ([this post](https://github.com/TeamCodeStream/codestream/issues/642#issuecomment-921626806) by marcelofarias).
